driver/src/orderbook/streamed/state.rs

```diff
@@ -33,8 +45,31 @@ struct LastSolution {
 }
 
 impl State {
-    /// Can be used to create a `crate::models::AccountState`.
-    pub fn account_state(
+    /// Errors if State has only received a partial solution which would make the result
+    /// inconsistent.
+    /// Errors if the batch is `Future` but is not actually in the future.
+    /// Account balances that overflow a U256 are skipped.
+    pub fn orderbook(
+        &self,
+        batch: Batch,
+    ) -> Result<(
+        impl Iterator<Item = ((UserId, TokenId), U256)> + '_,
+        impl Iterator<Item = ModelOrder> + '_,
```

Comment:
Is this a tuple of results or either or?

Reply:
Because the Result is on the outside it is a Result of a tuple. So either you have both elements of the tuple or an error.